### PR TITLE
DEV-4326: write_file

### PR DESCRIPTION
**Pull request by MSilva95**
> Before starting, please choose the relevant pull request **Labels**, **Reviewers**, and **Assignees**

### Why?

> Clearly state the reason for this change. What problem is it solving or feature is it adding?

### Solution Overview

> Provide an overview of the solution implemented in this pull request. This should be a high level overview without getting into technical details. If applicable, include screenshots of UI or use GitHub compliant [`mermaid`](https://mermaid.live/) graphs to visually represent the solution.

### Implementation Details

> Explain the details of the implementation and the reasoning behind it. What alternative approaches were considered and why was this approach chosen?

### CODE REVIEW

File: subjects/devops/write_file/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+## Write file
+
+### Instructions
+
+Create a file `write_file.py` which will have a function `to_do(input)` with one input a `list[tuple[dt.date, str]]` which will be a to do list example:
```
Comment (nprimo) [inline subjects/devops/write_file/README.md:5]: Nothing major - my spell checker suggest to spell to do list as follows: `to-do list`

File: subjects/devops/write_file/README.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+```python
+[
+    (dt.date(2022, 6, 1), "fix the seat"),
+    (dt.date(2022, 6, 2), "take the trash"),
+]
+```
+
+Your function needs to format and write the input in a text file `output.txt` as follow:
```
Comment (nprimo) [inline subjects/devops/write_file/README.md:14]: `as follow` should be `as follows`

File: subjects/devops/write_file/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+The `write()` method is used to write the string "Hello World!" to the file.
+
+- Make sure to include a newline character `\n` at the end of each line of output, to separate the tasks in the output file.
+
+- Test your function with different input formats, different date and task length, different date range and make sure that it works as expected.
+
+### References
+
+- [open a file](https://www.w3schools.com/python/python_file_write.asp)
```
Comment (nprimo) [inline subjects/devops/write_file/README.md:85]: Just a suggestion: I would give a link that showcases the `with open('file.txt') as file: ...` construct. For example, this one https://docs.python.org/3/tutorial/inputoutput.html#tut-files

File: subjects/devops/write_file/README.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+Your function needs to format and write the input in a text file `output.txt` as follow:
+
+```bash
+$ cat output.txt
+Wednesday 01 June 2022: fix the seat
+Thursday 02 June 2022: take the trash
+$
+```
+
+Make sure you follow the correct format for the output.
```
Comment (nprimo) [inline subjects/devops/write_file/README.md:23]: What should happen when the list passed is empty?

What should happen if the format of the list is not the one required? Should the student always assume that the list is correctly formatted?

**Review by nprimo — CHANGES_REQUESTED** [review]
I left some suggestions inline

File: subjects/devops/write_file/README.md (hunk 1 above)
Comment (MSilva95) [reply subjects/devops/write_file/README.md:5]: done

File: subjects/devops/write_file/README.md (hunk 2 above)
Comment (MSilva95) [reply subjects/devops/write_file/README.md:14]: done

File: subjects/devops/write_file/README.md (hunk 3 above)
Comment (MSilva95) [reply subjects/devops/write_file/README.md:85]: Done, thanks for the link!

File: subjects/devops/write_file/README.md (hunk 4 above)
Comment (MSilva95) [reply subjects/devops/write_file/README.md]: i will add clarification on the readme, but yes they should assume the list is correct.

**Review by nprimo — APPROVED** [review]
LGTM! :rocket:

**Pull request merged**